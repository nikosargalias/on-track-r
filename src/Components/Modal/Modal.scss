@import '../../Styles/abstracts.scss';

.overlay {
    z-index: 10;
    position: fixed;
    top: 0;
    left: 0;
    min-height: 100%;
    width: 100vw;
    background: rgba(0, 0, 0, 0.8);
    padding: 2rem;
}

.modal {
    display: flex;
    align-items: center;
    justify-content: center;

    &__content {
        position: relative;
        display: flex;
        align-items: flex-start;
        padding: 2.5rem 2rem;
        background: $bg-primary;
        border-radius: 5px;
    }

    &__img {
        margin-right: 2rem;
    }

    &__close {
        background: transparent;
        color: white;
        background: #1d1d1d;
        border-radius: 50px;
        padding: 0.2rem;
        height: 2rem;
        width: 2rem;
        position: absolute;
        top: 0.5rem;
        right: 1rem;
        cursor: pointer;
    }
}
