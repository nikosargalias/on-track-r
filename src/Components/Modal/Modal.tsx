import React, { useCallback, useEffect, useRef } from 'react';
import { BookType } from '../../Shared/Types/BookType';
import './Modal.scss';

const Modal = ({
    book,
    handleClose,
    focusElem,
}: {
    book: BookType;
    handleClose: () => void;
    focusElem?: HTMLElement | undefined;
}) => {
    const buttonRef = useRef<HTMLButtonElement>(null);
    const handleClickOnOverlay = useCallback(
        (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
            const target = e.target as HTMLDivElement;
            if (target.id !== 'modal-overlay') {
                return;
            }
            handleClose();
        },
        []
    );

    useEffect(() => {
        if (buttonRef.current) {
            buttonRef.current.focus();
        }

        return () => {
            if (focusElem) {
                focusElem.focus();
            }
        };
    }, [buttonRef]);
    return (
        <div
            className='overlay modal'
            id='modal-overlay'
            onClick={handleClickOnOverlay}
        >
            <article
                className='modal__content js-modal__content'
                tabIndex={0}
                ref={buttonRef}
            >
                <div className='book__info'>
                    <h2>Title: {book?.book_title}</h2>
                    <p>Author: {book?.book_author}</p>
                    <p>Publication city: {book?.book_publication_city}</p>
                    <p>Publication country: {book?.book_publication_country}</p>
                    <p>Publication year: {book?.book_publication_year}</p>
                </div>
                <button
                    aria-label='Close popup'
                    onClick={handleClose}
                    className='modal__close'
                >
                    &#10005;
                </button>
            </article>
        </div>
    );
};

export default Modal;
