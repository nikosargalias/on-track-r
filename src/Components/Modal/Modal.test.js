import React from 'react';
import renderer from 'react-test-renderer';
import Modal from './Modal';

it('renders correctly', () => {
    const Movie = {
        Title: 'string',
        Year: 'string',
        imdbID: 'string',
        Type: 'string',
        Poster: 'string',
    };
    const handleClose = () => {};
    const tree = renderer
        .create(<Modal movie={Movie} handleClose={handleClose} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
