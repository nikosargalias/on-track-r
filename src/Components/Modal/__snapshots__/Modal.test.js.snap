// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="overlay modal"
  id="modal-overlay"
  onClick={[Function]}
>
  <article
    className="modal__content js-modal__content"
  >
    <img
      className="modal__img"
      src="string"
    />
    <div
      className="movie__info"
    >
      <h2>
        string
        <span>
          string
        </span>
      </h2>
      <p>
        No Director information available
      </p>
      <p>
        No cast information available
      </p>
      <p>
        No movie description available
      </p>
    </div>
    <button
      aria-label="Close popup"
      className="modal__close"
      onClick={[Function]}
    >
      âœ•
    </button>
  </article>
</div>
`;
